### DISCUSSION TITLE
Latest time directory not being written out to DAFoam correctly

### DISCUSSION THREAD

**Discussion by uttamcadambi07** [mdolab/dafoam#539] [General]
Hi,

I am facing a rather peculiar issue. 

In dafoam v3.0.3,
Suppose I have a start and end time of 0 and 5000 respectively for a steady state solver and the write interval is 500 (so I get directories of 500,1000,1500,2000... upto convergence or 5000 whichever comes first).

Now if my solver tolerance does not fall below the residual tolerance that is set in the runScript.py, DAFoam takes the last time directory (please see the message below from my log file)
```
Moving time 5000 to 0.00000001
Running adjoint Solver 001
```
Attaching
All good. 

But now in dafoam v3.0.6

When I increase the residual tolerance to say 5 times the previous value (earlier it was 1e-05, now i make it 5e-05), i get the convergence at a much earlier step (say 2636, find the log below).
```
Time = 2636
Minimal residual 4.949060261950095e-05 satisfied the prescribed tolerance 5e-05

FI-Ux-fieldInversion: 1.000000278857031
Printing Primal Residual Statistics.
U Residual Norm2: (0.005937347181784317 0.001033145674242338 8.099734026025834e-17)
U Residual Mean: (3.726968453789449e-06 9.221305162354941e-07 1.410668903881984e-19)
U Residual Max: (0.001165196917584897 0.000303348489606229 6.015791634229455e-18)
p Residual Norm2: 0.03148081653726072
p Residual Mean: 6.340726006515971e-06
p Residual Max: 0.01178775464837706
nuTilda Residual Norm2: 0.0002074931750870586
nuTilda Residual Mean: 7.425390188392298e-08
nuTilda Residual Max: 3.365538814551318e-05
phi Residual Norm2: 0.0004883509413702177
phi Residual Mean: 8.198543199793559e-08
phi Residual Max: 7.395522024217564e-05
End

```

But, in this case, DAFoam takes the first time directory (500) and not the latest (please find below). 
```
+--------------------------------------------------------------------------+
|              Evaluating Objective Function Sensitivities 001             |
+--------------------------------------------------------------------------+
writeDeformedFFDs is deprecated since v3.0.1!
Moving time 500 to 0.00000001
Running adjoint Solver 001
dRdWT Jacobian Free created!

```

This means that my adjoint solver does not convergence (since obviously the solution at 500 is not converged enough). Why is this happening?

**Comment by uttamcadambi07** [comment]
One workaround that I am trying is to set the writeinterval  to be the same as the end time.

**Reply by friedenhe** [reply]
Yes, that is the right setting to use. You are not supposed to write multiple solutions in one flow simulation.
